Refactor Likes component and utils.js

shakilakamalasena opened the pull request:
- Update Likes component to include a button to clear all liked images
- Add clearFavorites function to utils.js to remove all liked images from local storage

shakilakamalasena committed:
Refactor Likes component and utils.js

- Update Likes component to include a button to clear all liked images
- Add clearFavorites function to utils.js to remove all liked images from local storage

JoarHansson commented on src/pages/Likes/Likes.jsx:
```diff
@@ -19,6 +20,13 @@ const Likes = () => {
     setFavorites(getFavoritesFromLocalStorage());
   };
 
+  const handleClear = () => {
+    if (window.confirm("Are you sure you want to clear all likes?")) {
+      clearFavorites();
+      setFavorites(getFavoritesFromLocalStorage());
+    }
+  };
```
This works well but in the future the window.confirm method can be swapped out to something that fits the UI better, like a custom modal. I will create a separate issue about this!

shakilakamalasena replied:
Okay!

JoarHansson commented on src/pages/Likes/Likes.jsx:
```diff
@@ -38,10 +46,16 @@ const Likes = () => {
         )}
       </div>
 
+      <button onClick={handleClear} className={styles.clearButton}>
+        Clear Likes
+      </button>
```
This works but doesn't fit the UI. 

There is however a Button component in the project that you can import and use here. The classname clearButton will not be needed since the Button component is styled already. 

Also, in your implementation the button is visible even if there are no images to clear. If there are no favorites in local storage the button shouldn't be visible for the user.

A solution would be to conditionally render the button based on if `favorites` (`const [favorites, setFavorites] = useState(getFavoritesFromLocalStorage());` is empty or not.

If you could make these changes I will merge your PR. 

Thanks!

shakilakamalasena replied:
I'll do these changes.

JoarHansson commented on src/services/utils.js:
```diff
@@ -51,3 +51,7 @@ export const removeFromFavorites = (itemToRemove) => {
   localStorage.setItem("favorites", JSON.stringify(favorites));
 };
+
+export const clearFavorites = () => {
+  localStorage.removeItem("favorites");
+};
```
Very good, clear and concise.

shakilakamalasena replied:
Thank you!

JoarHansson reviewed (changes requested):
Thanks for your PR!

I wrote some comments about a few changes that would be nice to have done before the PR is merged.

shakilakamalasena committed:
Fix: Conditionally render 'Clear Likes' button and display 'No likes' message

- Ensure the 'Clear Likes' button is only displayed when there are favorites.
- Display "You haven't liked any images yet..." message when there are no liked images.
- Removed unnecessary class for the button as it's already styled.

shakilakamalasena commented:
I added Conditionally render 'Clear Likes' button and display 'No likes' message as requested.

JoarHansson reviewed (approved):
With the changes done I think everything looks good, and I will merge the PR.

Thank you for your contributions!

shakilakamalasena commented:
Thank you for reviewing and merging the PR! I'm glad the changes look good. Let me know if there's anything else I can help with in the future!
